server/src/main/java/org/elasticsearch/discovery/HandshakingTransportAddressConnector.java

```diff
@@ -101,7 +100,6 @@ public void onResponse(DiscoveryNode remoteNode) {
                             try {
                                 // success means (amongst other things) that the cluster names match
                                 logger.trace("[{}] handshake successful: {}", transportAddress, remoteNode);
-                                IOUtils.closeWhileHandlingException(connection);
```

Comment:
Hmm I think we don't track this connection anywhere so if we don't close it here then it just leaks AFAICT. Am I missing something obvious?

Reply:
Umm no you're not my bad missed reverting that line. Weird though that this didn't fail any tests?

Reply:
We just let connections die naturally in tests today, I don't think there's anything that would catch this.